src/Equations/elastic/Kernels/GravitationalFreeSurfaceBC.h

```diff
@@ -127,103 +112,82 @@ class GravitationalFreeSurfaceBc {
     alignas(ALIGNMENT) real dofsFaceNodalStorage[tensor::INodal::size()];
     auto dofsFaceNodal = init::INodal::view::create(dofsFaceNodalStorage);
 
-    auto f = [&](
-        ode::ODEVector& du,
-        ode::ODEVector& u,
-        double time) {
-      // Evaluate Taylor series at time
+    // Temporary buffer to store nodal face coefficients at some time t
+    alignas(ALIGNMENT) std::array<real, nodal::tensor::nodes2D::Shape[0]> prevCoefficients;
+
+    const double deltaT = timeStepWidth;
+    const double deltaTInt = timeStepWidth;
+
+    // Initialize first component of Taylor series
+    for (unsigned int i = 0; i < nodal::tensor::nodes2D::Shape[0]; ++i) {
+      prevCoefficients[i] = rotatedFaceDisplacement(i, 0);
+      // This is clearly a zeroth order approximation of the integral!
+      integratedDisplacementNodal(i) = deltaTInt * rotatedFaceDisplacement(i, 0);
+    }
+
+    // Coefficients for Taylor series
+    double factorEvaluated = 1;
+    double factorInt = deltaTInt;
+
+    auto* derivativesOffsets = timeKernel.getDerivativesOffsets();
+    projectKernel.INodal = dofsFaceNodal.data();
+    for (unsigned i = 0; i < yateto::numFamilyMembers<tensor::dQ>(); ++i) {
+      projectKernel.dQ(i) = derivatives + derivativesOffsets[i];
+    }
+
+#ifdef USE_ELASTIC
+    const double rho = materialData.local.rho;
+    const double g = getGravitationalAcceleration(); // [m/s^2]
+    const double Z = std::sqrt(materialData.local.lambda * rho) ;
+#endif
```

Comment:
What happens else? Should abort, right?

Reply:
It is a standard free surface boundary condition then - this is just s.t. it compiles when using other equations. I'll remove that at some point, but for now I blame our code structure.